program/clientcommon/runtime.go

```diff
@@ -57,13 +58,23 @@ func InitRuntime(binpath, sockpath, dbpath string) *eval.Evaler {
 	}
 	var cl *daemonapi.Client
 	if sockpath != "" && dbpath != "" {
+		sockstate, statErr := os.Stat(sockpath)
 		cl = daemonapi.NewClient(sockpath)
-		_, statErr := os.Stat(sockpath)
 		killed := false
 		if statErr == nil {
 			// Kill the daemon if it is outdated.
 			version, err := cl.Version()
 			if err != nil {
+				if sockstate != nil {
+					sockstatesys := sockstate.Sys()
+					if sockstatesys != nil {
+						err := os.Remove(sockpath)
+						if err != nil {
+							fmt.Fprintln(os.Stderr, "warning: cannot remove socket : ", err)
+						}
+						goto spawnDaemonStart
```

Comment:
Avoid `goto`.

Reply:
Could I instead of using goto recall upon the function itself as in `InitRuntime(toSpawn.BinPath, toSpawn.SockPath, toSpawn.DbPath)` ?
Trying to spawn the daemon directly without going through all the steps causes crashes to sh because of a null pointer to the non-existent sock file . Doing it this way seems to work quite okay , even though I'm thinking that maybe this can also be pointed towards whenever the client realizes it needs  a connection with the daemon and the daemon is dead . 
The way I'm handling it right now , it basically restarts the daemon on starting a new client if the current one is not responding .